src/ScheduleFunctions.cpp

```diff
@@ -1058,9 +1056,8 @@ class InjectFunctionRealization : public IRMutator {
         vector<pair<string, Expr>> containers;
         while (1) {
             if (const LetStmt *l = body.as<LetStmt>()) {
-                const Call *call = l->value.as<Call>();
-                if (!(call && call->is_intrinsic(Call::promise_clamped)) &&
-                    !is_pure(l->value)) {
+                const Call *call = Call::as_intrinsic(l->value, {Call::promise_clamped});
```

Comment:
Some of these variable names are misleading now when you just read the if condition. How about:

call -> promise_clamped